File: IRPlayer-swift/Class/Core/FFPlayer/FFmpeg/Frame/IRFFCVYUVVideoFrame.m
//
//  IRFFCVYUVVideoFrame.m
//  IRPlayer
//
//  Created by Phil on 2019/10/25.
//  Copyright © 2019 Phil. All rights reserved.
//

#import "IRFFCVYUVVideoFrame.h"

@implementation IRFFCVYUVVideoFrame {
    BOOL shouldRelease;
}

- (IRFFFrameType)type
{
    return IRFFFrameTypeCVYUVVideo;
}

- (instancetype)initWithAVPixelBuffer:(CVPixelBufferRef)pixelBuffer shouldRelease:(BOOL)shouldRelease
{
    if (self = [super init]) {
        self->_pixelBuffer = pixelBuffer;
        self->shouldRelease = shouldRelease;
    }
    return self;
}

- (int)width {
    const GLsizei frameWidth = (GLsizei)CVPixelBufferGetWidth(self->_pixelBuffer);
    return frameWidth;
}

- (int)height {
    const GLsizei frameHeight = (GLsizei)CVPixelBufferGetHeight(self->_pixelBuffer);
    return frameHeight;
}

- (void)dealloc
{
    if (self->_pixelBuffer) {
        if (shouldRelease) {
            CVPixelBufferRelease(self->_pixelBuffer);
        }
        self->_pixelBuffer = NULL;
    }
}

@end

File: IRPlayer-swift/Class/Core/FFPlayer/FFmpeg/Frame/IRFFCVYUVVideoFrame.swift
//
//  IRFFCVYUVVideoFrame.swift
//  IRPlayer-swift
//
//  Created by irons on 2024/7/26.
//

import Foundation
import CoreVideo

@objcMembers public class IRFFCVYUVVideoFrame: IRFFVideoFrame {
    public let pixelBuffer: CVPixelBuffer

    init(pixelBuffer: CVPixelBuffer) {
        self.pixelBuffer = pixelBuffer
        super.init()
        width = CVPixelBufferGetWidth(pixelBuffer)
        height = CVPixelBufferGetHeight(pixelBuffer)
    }

    override var type: IRFFFrameType {
        return .cvyuvVideo
    }

//    override var width: Int {
//        return CVPixelBufferGetWidth(pixelBuffer)
//    }
//
//    override var height: Int {
//        return CVPixelBufferGetHeight(pixelBuffer)
//    }
}

